src/plugins/vis_augmenter/public/utils/utils.ts

```diff
@@ -14,14 +14,23 @@ import {
 import { ISavedAugmentVis, SavedAugmentVisLoader, VisLayerFunctionDefinition } from '../';
 
 export const isEligibleForVisLayers = (vis: Vis, dimensions: VislibDimensions): boolean => {
-  const isDateHistogram =
-    vis.data.aggs?.byTypeName('date_histogram').length === 1 && vis.data.aggs?.aggs.length === 2;
+  // Only support date histogram and ensure there is only 1 x-axis and it has to be on the bottom
+  const isValidXaxis =
+    vis.data.aggs?.byTypeName('date_histogram').length === 1 &&
+    vis.params.categoryAxes.length === 1 &&
+    vis.params.categoryAxes[0].position === 'bottom';
+  // Support 1 segment for x axis bucket (that is date_histogram) and support metrics for
+  // multiple supported yaxis only. If there are other aggregation types, this is not
+  // valid for augmentation
+  const hasCorrectAggregationCount =
+    vis.data.aggs !== undefined &&
+    vis.data.aggs?.bySchemaName('metric').length === vis.data.aggs?.aggs.length - 1;
```

Comment:
Should we also check that there is >= 1 metric agg? Or should we still support 0?

Reply:
That is a fair point, there is no need to support a graph that will not have any metrics since there would be no reason to overlay events on the graph.